.github/workflows/publish-docker-pr.yml

```diff
@@ -0,0 +1,54 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: publish-docker
+
+on:
+  pull_request:
+
+env:
+  HUB: ghcr.io/apache/skywalking-banyandb
+
+jobs:
+  build:
+    if: github.repository == 'apache/skywalking-banyandb'
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      packages: write
+    timeout-minutes: 90
+    strategy:
+      fail-fast: true
+    env:
+      TAG: ${{ github.sha }}
+    steps:
+      - name: Install Go
+        uses: actions/setup-go@v2
+        with:
+          go-version: 1.16
+      - name: Install goimports
+        run: go install golang.org/x/tools/cmd/goimports@latest
+      - name: Check out code into the Go module directory
+        uses: actions/checkout@v2
+      - name: Update dependencies 
+        run: GOPROXY=https://proxy.golang.org go mod download
+      - name: Generate codes
+        run: make generate
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v1
```

Comment:
I'm very confused why we need so many steps out of Makefile, instead of making them automatically run in Makefile by using dependent tasks so that developers only need to run `make docker` without caring about the GitHub workflow file. The same concern exists in the other workflow file `.github/workflows/publish-docker-main.yml`

Reply:
I suppose we can move `goimport` installation into Makefile and have a better dependency tree

Reply:
> I suppose we can move `goimport` installation into Makefile and have a better dependency tree

Not only `goimport`, but also `make generate`